src/error.rs

```diff
@@ -7,9 +7,13 @@ pub type Result<T> = result::Result<T, Error>;
 pub type FatalResult<T> = result::Result<T, Fatal>;
 
 #[must_use]
+#[derive(Debug)]
 pub enum Fatal {
     /// Output file write error.
     Output(io::Error),
+
+    /// An unrecoverable internal error.
+    InternalError,
```

Comment:
Maybe I should rename it to `AssertionFailed`? And then provide more documentation mentioning it exists for better integration into the diagnostics systems and to provide a proper, non-panicking error handling program flow.

Reply:
I think `InternalCompilerError` is fine here, because that's literally what it is. We print a diagnostic for one, so why shouldn't we return it as one? :)

Reply:
I just find it a bit confusing labeling it a *compiler* error. That may be familiar for programmers but could be entirely confusing (in an already uncomfortable situation of a crash) to an other user demographic of which there are plenty others outside of programmers. But possibly fine?

Reply:
I mean in the `Display` impl we can still just have `"\n\nCan not continue due to internal error"`, which is the actual user-facing interface.
Also in the `Diagnostics` it's marked as `internal compiler error: $Message`.

Reply:
Right. I'm changing it to `InternalCompilerError` as proposed.